### ISSUE TITLE
[FEATURE REQUEST]The Issue Regarding OIIO Support for ARW File Format

### ISSUE THREAD

**Issue by cool2528** [AcademySoftwareFoundation/OpenImageIO#4117]
When I use the OIIO library to read ARW file format, it shows an error indicating unsupported file format or not a valid raw file format. However, upon checking, I found that LibRaw does support parsing this format. Therefore, I hope to see if it's possible to add support for parsing ARW file format.

![image](https://github.com/AcademySoftwareFoundation/OpenImageIO/assets/42990446/b76bf521-d3b3-4614-9611-1a67dec6017d)
I have uploaded a sample file in ARW file format, hoping it will be helpful for analyzing this file format.

**Comment by lgritz** [comment]
All formats that libraw reads are the same to OIIO; there's no place in the code where we include or exclude individual formats that libraw supports. 

Can you tell us the exact error message?
(And is there an arw file you can point us at, to give it a try on our end?)

**Comment by cool2528** [comment]
@lgritz 
Sorry, just found out there's a size limit for uploading files on GitHub. Could you please provide your email so I can send it to you privately? Also, please do not disclose this file, as it is the client's confidential document.

**Comment by lgritz** [comment]
lg@openimageio.org ought to get to me. Anything you send will be 100% confidential and be deleted as soon as I've identified the problem.

**Comment by cool2528** [comment]
@lgritz Hello, the required ARW file sample has been sent to your email. Please check your inbox. My email has the prefix "jmp" and is a Google email.

**Comment by lgritz** [comment]
Works for me, opens fine with OIIO on m Mac using libraw 0.21.2 and OIIO master.

Let's take a step back. Can you tell us the precise error you got, and the command you tried?

**Comment by cool2528** [comment]
@lgritz The version of oiio I use is #define OIIO_VERSION_MAJOR 2
#define OIIO_VERSION_MINOR 4
#define OIIO_VERSION_PATCH 13
#define OIIO_VERSION_TWEAK 0 has_error() returns true when calling ImageBuf to read The compiled libraw supports DNGSDK I'm not sure if it's the compiled libraw that's the problem?

**Comment by cool2528** [comment]
The version of oiio I'm using is. When calling `ImageBuf` to read, `has_error()` returns true. The compiled libraw supports DNGSDK. I'm not sure if there are any issues with the compiled libraw.
#define OIIO_VERSION_MAJOR 2
#define OIIO_VERSION_MINOR 4
#define OIIO_VERSION_PATCH 13
#define OIIO_VERSION_TWEAK 0

**Comment by lgritz** [comment]
has_error() returns true, but does geterror() return any useful message that gives us a clue?

What if we take your app out of the equation for the moment, does it work with a pure OIIO codebase pipeline? Like

    oiiotool --info -v -stats yourfile.arw

does it work, or does it give an error?

Also,

    oiiotool --help 

should print the library versions (near the bottom, where it lists "Dependent libraries:"). Can you verify that libraw is there, and what version it's using?

**Comment by cool2528** [comment]
@lgritz Sorry, it might be due to issues with the compilation of the libraw library on my end during debugging. I apologize. Are there any specific things to pay attention to when compiling the libraw library for use with oiio?
![1705751042121](https://github.com/AcademySoftwareFoundation/OpenImageIO/assets/42990446/49b99ab2-8c67-48db-b67a-6b5027290d9c)

**Comment by cool2528** [comment]
@lgritz  I think I've found the answer. Support for the arw file format in the libraw library began with version 0.20.2. The oiio library I compiled relies on the source code of libraw version 0.20.2. I apologize for the inconvenience, but at least finding the reason is always a positive outcome. I'm sharing this information in the hope that it may serve as a reference for others who encounter the same issue in the future.

These are the results of my program tested with libraw version 0.20.2.
![image](https://github.com/AcademySoftwareFoundation/OpenImageIO/assets/42990446/f460a23b-71c6-49fe-a0dd-3d1a8a011950)
